dotnet/src/Connectors/Connectors.OpenAIV2/Core/ClientCore .AudioToText.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+/* 
+Phase 02
+
+- This class was created focused in the Image Generation using the SDK client instead of the own client in V1.
+- Added Checking for empty or whitespace prompt.
+- Removed the format parameter as this is never called in V1 code. Plan to implement it in the future once we change the ITextToImageService abstraction, using PromptExecutionSettings.
+- Allow custom size for images when the endpoint is not the default OpenAI v1 endpoint.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Text;
+using OpenAI.Audio;
+
+namespace Microsoft.SemanticKernel.Connectors.OpenAI;
+
+/// <summary>
+/// Base class for AI clients that provides common functionality for interacting with OpenAI services.
+/// </summary>
+internal partial class ClientCore
+{
+    /// <summary>
+    /// Generates an image with the provided configuration.
+    /// </summary>
+    /// <param name="input">Input audio to generate the text</param>
+    /// <param name="executionSettings">Audio-to-text execution settings for the prompt</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Url of the generated image</returns>
+    internal async Task<IReadOnlyList<TextContent>> GetTextFromAudioContentsAsync(
+        AudioContent input,
+        PromptExecutionSettings? executionSettings,
+        CancellationToken cancellationToken)
+    {
+        if (!input.CanRead)
+        {
+            throw new ArgumentException("The input audio content is not readable.", nameof(input));
+        }
+
+        OpenAIAudioToTextExecutionSettings? audioExecutionSettings = OpenAIAudioToTextExecutionSettings.FromExecutionSettings(executionSettings);
+        AudioTranscriptionOptions? audioOptions = AudioOptionsFromExecutionSettings(executionSettings);
+
+        Verify.ValidFilename(audioExecutionSettings?.Filename);
+
+        var memoryStream = new MemoryStream(input.Data!.Value.ToArray());
+
+        AudioTranscription responseData = (await RunRequestAsync(() => this.Client.GetAudioClient(this.ModelId).TranscribeAudioAsync(memoryStream, audioExecutionSettings?.Filename, audioOptions)).ConfigureAwait(false)).Value;
```

Comment:
You are using RunRequestAsync because TranscribeAudioAsync can run synchronously and you want to force async? I don't fully understand purpose of this method RunRequestAsync

Reply:
This method serves as a wrapper to capture exceptions property if they happen.

```csharp
 private static async Task<T> RunRequestAsync<T>(Func<Task<T>> request)
 {
     try
     {
         return await request.Invoke().ConfigureAwait(false);
     }
     catch (ClientResultException e)
     {
         throw e.ToHttpOperationException();
     }
 }
```